src/main/java/com/diffmin/patch/PatchGeneration.java

```diff
@@ -23,6 +24,8 @@
     private final List<Pair<CtElement, CtElement>> updatePatches = new ArrayList<>();
     private final List<ImmutableTriple<Integer, CtElement, CtElement>> insertPatches =
             new ArrayList<>();
+    private final List<Pair<CtElement, ImmutableTriple<Integer, CtElement, CtElement>>>
```

Comment:
Since `MoveOperation` is a combination of `Insert` and `Delete`, I combined the data structure accordingly.

Reply:
This is getting rather hard to read. It might be time to create dedicate container types for the different operations. Like you had in the beginning, but now it's actually warranted.

For another PR of course, just a thought.